### PR TITLE
Modified Factorio Egg for better start script

### PR DESCRIPTION
**Pull request by IConnC**
Originally, the startup script for the Factorio egg would fail if the save file was missing, this PR fixes that by running the save generator prior to server start if the save file does not already exist.

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### Changes to an existing Egg:

1. [X] Have you added an explanation of what your changes do and why you'd like us to include them?
2. [X] Have you tested your Egg changes?

### CODE REVIEW

File: game_eggs/factorio/factorio/egg-factorio.json [hunk 1]
```diff
@@ -3,21 +3,25 @@
...
-    "exported_at": "2020-06-26T12:55:32-04:00",
+    "exported_at": "2022-01-02T14:45:19-05:00",
     "name": "Factorio",
     "author": "parker@parkervcp.com",
     "description": "The vanilla Factorio server.\r\n\r\nhttps:\/\/www.factorio.com\/",
-    "image": "quay.io\/parkervcp\/pterodactyl-images:base_debian",
-    "startup": ".\/bin\/x64\/factorio --port {{SERVER_PORT}} --server-settings data\/server-settings.json --start-server {{SAVE_NAME}}.zip",
+    "features": null,
+    "images": [
+        "quay.io\/parkervcp\/pterodactyl-images:base_debian"
```
Comment (gOOvER) [inline game_eggs/factorio/factorio/egg-factorio.json:12]: Pls use the new yolks

File: game_eggs/factorio/factorio/egg-factorio.json [hunk 2]
```diff
@@ -3,21 +3,25 @@
...
             "container": "debian:buster-slim",
```
Comment (gOOvER) [inline game_eggs/factorio/factorio/egg-factorio.json:25]: as here: pls use the new yolks

**Review by gOOvER — CHANGES_REQUESTED** [review]
quay is outdated. You should add the new yolk images

**Comment by IConnC** [comment]
Updated to ghcr.io repository

**Review by gOOvER — APPROVED** [review]
for me it works. Server was installed and i can connect

**Review by parkervcp — APPROVED** [review]

**Pull request merged**